src/AutorestSwift/Models/Complex/Parameter.swift

```diff
@@ -60,4 +60,18 @@ class Parameter: Value {
 
         try super.encode(to: encoder)
     }
+
+    // MARK: CustomDebugStringConvertible
+
+    public var debugDescription: String {
+        return debugString() ?? description
+    }
+}
+
+extension Parameter: Equatable {
+    static func == (lhs: Parameter, rhs: Parameter) -> Bool {
```

Comment:
Question: When do we need to == Parameter?

Reply:
To use `params.contains(param)` it requires that `ParameterType` conform to `Equatable`, and thus `Parameter` must as well.

Reply:
Otherwise I was just using a dictionary to simulate the uniqueness, but that destroyed the ordering (as would using `Set<T>`)